### PR TITLE
ignore errors if ignore saved search flag is true

### PR DESCRIPTION
**Pull request by vishalxebia**

### CODE REVIEW

File: utils/splunk/splunk_telemetry_base.py [hunk 1]
```diff
@@ -55,8 +55,10 @@ def check(self, instance):
                 self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK)
         except Exception as e:
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=instance.tags, message=str(e))
-            self.log.exception("Splunk event exception: %s" % str(e))
-            raise CheckException("Error getting Splunk data, please check your configuration. Message: " + str(e))
+            if not instance.instance_config.ignore_saved_search_errors:
+                self.log.exception("Splunk event exception: %s" % str(e))
+                raise CheckException("Error getting Splunk data, please check your configuration. Message: " + str(e))
+            self.log.warning("An exception occured in splunk telemetry and ignored as ignore_saved_search_errors flag is true.")
```
Comment (whazenberg) [inline utils/splunk/splunk_telemetry_base.py]: can we show the actual exception, or at least the message, as well?

Or move line 59, self.log.exception, outside of the if-structure?

File: utils/splunk/splunk_telemetry_base.py [hunk 2]
```diff
@@ -81,8 +83,10 @@ def _dispatch_and_await_search(self, instance, saved_searches):
                 self.update_persistent_status(instance.instance_config.base_url, saved_search.name, sid, "add")
                 search_ids.append((sid, saved_search))
             except FinalizeException as e:
-                self.log.error("Got an error %s while finalizing the saved search %s" % (e.message, saved_search.name))
-                raise e
+                if not instance.instance_config.ignore_saved_search_errors:
+                    self.log.error("Got an error %s while finalizing the saved search %s" % (e.message, saved_search.name))
+                    raise e
+                self.log.warning("Ignoring the finalize exception as ignore_saved_search_errors flag is true")
```
Comment (whazenberg) [inline utils/splunk/splunk_telemetry_base.py:89]: same here. Display the actual error?

**Review by whazenberg — APPROVED** [review]

File: utils/splunk/splunk_telemetry_base.py (hunk 1 above)
Comment (vishalxebia) [reply utils/splunk/splunk_telemetry_base.py]: done.

File: utils/splunk/splunk_telemetry_base.py (hunk 2 above)
Comment (vishalxebia) [reply utils/splunk/splunk_telemetry_base.py:89]: done.

**Review by whazenberg — APPROVED** [review]

**Pull request merged**